src/config/ConfigScope.ts

```diff
@@ -180,6 +182,41 @@ export class ConfigScope {
     return rawValue === 'true'
   }
 
+  getMandatoryJsonObject<T extends object>(param: string, schema: ZodSchema<T>): T {
+    const rawValue = this.getMandatory(param)
+    const parsedValue = schema.safeParse(JSON.parse(rawValue))
+
+    if (!parsedValue.success) {
+      throw new InternalError({
+        message: `Configuration parameter ${param} must be a valid JSON meeting the given schema, but was ${rawValue}`,
+        errorCode: 'CONFIGURATION_ERROR',
+      })
+    }
+
+    return parsedValue.data
+  }
+
+  getOptionalJsonObject<T extends object>(param: string, schema: ZodSchema<T>, defaultValue: T): T {
+    const rawValue = this.getOptional(param, '{}')
+    const parsedValue = schema.safeParse(JSON.parse(rawValue))
+
+    return parsedValue.success ? parsedValue.data : defaultValue
```

Comment:
I would say that optional method should still throw if we fail the validation on passed object, it should only fallback to default if there is no value at all, or value is empty string.

Reply:
Fair point, that is a more reasonable behaviour, thank you so much Igor 🙌